src/components/photo_gallery.js

```diff
@@ -0,0 +1,145 @@
+import _ from 'lodash';
+import React, {Component, PropTypes} from 'react';
+import shouldComponentUpdate from 'react-addons-shallow-compare';
+
+import Row from 'muicss/lib/react/row';
+import Col from 'muicss/lib/react/col';
+import GridList from 'material-ui/lib/grid-list/grid-list';
+import RaisedButton from 'material-ui/lib/raised-button';
+import FontIcon from 'material-ui/lib/font-icon';
+
+import GalleryItem from './photo_gallery_item';
+
+export default class PhotoGallery extends Component {
+  static propTypes () {
+    return {
+      photos: PropTypes.array.isRequired,
+      editable: PropTypes.bool.isRequired,
+      onChange: PropTypes.func.isRequired,
+      showDialog: PropTypes.func
+    };
+  }
+
+  constructor () {
+    super();
+    this.state = {
+      photos: [],
+      selected: []
+    };
+
+    this.shouldComponentUpdate = shouldComponentUpdate.bind(this);
+
+    this.confirmDelete = this.confirmDelete.bind(this);
+    this.isSelected = this.isSelected.bind(this);
+    this.toggleSelected = this.toggleSelected.bind(this);
+    this.confirmBulkDelete = this.confirmBulkDelete.bind(this);
+  }
+
+  componentWillMount () {
+    this.setState({
+      photos: this.props.photos
+    });
+  }
+
+  isSelected (id) {
+    return this.state.selected.indexOf(id) >= 0;
+  }
+
+  toggleSelected (id) {
+    const {state} = this;
+
+    if (this.isSelected(id)) {
+      return this.setState({
+        ...state,
+        selected: _.without(state.selected, id)
+      });
+    }
+
+    this.setState({
+      ...state,
+      selected: [
+        ...state.selected,
+        id
+      ]
+    });
+  }
+
+  omitPhotos (ids) {
+    return _.filter(this.state.photos, (photo) => {
+      return ids.indexOf(photo.id) < 0;
+    })
+  }
+
+  confirmBulkDelete () {
+    const {state} = this;
+    const {showDialog, onChange} = this.props;
+    showDialog({
+      title: 'Confirm bulk deletion',
+      content: <div>Are you sure you want to delete <strong>{state.selected.length}</strong> images?</div>,
```

Comment:
Nice touch including the number of photos; it wasn't in the requirements, so great job!